fdbclient/SimpleConfigTransaction.actor.cpp

```diff
@@ -30,39 +30,40 @@
 
 class SimpleConfigTransactionImpl {
 	ConfigTransactionCommitRequest toCommit;
-	Future<Version> getVersionFuture;
+	Future<ConfigGeneration> getGenerationFuture;
 	ConfigTransactionInterface cti;
 	int numRetries{ 0 };
 	bool committed{ false };
 	Optional<UID> dID;
 	Database cx;
 
-	ACTOR static Future<Version> getReadVersion(SimpleConfigTransactionImpl* self) {
+	ACTOR static Future<ConfigGeneration> getGeneration(SimpleConfigTransactionImpl* self) {
```

Comment:
To me it seems like all these actors can be abstracted so both the simple config transaction and Paxos config transaction classes can use them, instead of repeating the logic in both places. The simple config transaction logic is the same as the Paxos config transaction logic except with one node instead of multiple. It does look like there are some specifics to each like the string in the `TraceEvent`s, but I thought I would point out that a lot of this code looks very similar between the two classes.

Reply:
The final fault tolerant `PaxosConfigTransaction::commit` and `PaxosConfigTransaction::getGeneration` implementations will look very different from the corresponding simple implementations, which is why they're implemented separately, even though the current implementations are very similar. There may be other places where we want to reduce duplicate code in the future, though.